### PR TITLE
Patchinfo component fix

### PR DESCRIPTION
**Pull request by saraycp**
Handle patchinfo tags as arrays. Otherwise, the loop crashes when there is only one element (not an array).
Include component specs.

Follow up https://github.com/openSUSE/open-build-service/pull/15257

### CODE REVIEW

File: src/api/app/components/patchinfo_component.html.haml [hunk 1]
```diff
@@ -36,7 +36,7 @@
     - if @patchinfo['releasetarget'].present?
       %h6 Targeted for release in the following projects
       %ul
-        - @patchinfo['releasetarget'].each do |releasetarget|
+        - [@patchinfo['releasetarget']].flatten.each do |releasetarget|
```
Comment (danidoni) [inline src/api/app/components/patchinfo_component.html.haml:39]: I would move all those `[@something].flatten` into the component class, expose it with a method or an instance variable

Comment (saraycp) [reply src/api/app/components/patchinfo_component.html.haml]: @danidoni  please have another look

File: src/api/spec/components/patchinfo_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe PatchinfoComponent, :vcr, type: :component do
```
Comment (eduardoj) [inline src/api/spec/components/patchinfo_component_spec.rb:3]:
```suggestion
RSpec.describe PatchinfoComponent, type: :component do
```
These specs seem not to need VCR.

Comment (saraycp) [reply src/api/spec/components/patchinfo_component_spec.rb]: True! Removed!

File: src/api/app/components/patchinfo_component.rb [hunk 3]
```diff
@@ -20,6 +20,22 @@ def initialize(patchinfo, path)
...
+    [@patchinfo['binary']].flatten
+  end
+
+  def packages
+    [@patchinfo['package']].flatten
+  end
+
+  def issues
+    [@patchinfo['issue']].flatten
+  end
```
[the remark this answers is not in the document]
Comment (saraycp) [reply src/api/app/components/patchinfo_component.rb]: Please have another look

**Review by eduardoj — APPROVED** [review]

**Pull request merged**